actions/class.ClientConfig.php

```diff
@@ -59,142 +65,160 @@ public function config()
         $this->setData('crossorigin', $this->isCrossorigin());
-        $this->setData('tao_base_www', $tao_base_www);
+        $this->setData('tao_base_www', $taoBaseWww);
 
         $this->setData('context', json_encode([
-            'root_url'       => ROOT_URL,
-            'base_url'       => $base_url,
-            'taobase_www'    => $tao_base_www,
-            'base_www'       => $base_www,
-            'base_lang'      => $lang,
-            'locale'         => $langCode,
-            'base_authoring_lang'  => $this->getUserLanguageService()->getAuthoringLanguage(),
-            'timeout'        => $this->getClientTimeout(),
-            'extension'      => $resolver->getExtensionId(),
-            'module'         => $resolver->getControllerShortName(),
-            'action'         => $resolver->getMethodName(),
+            'root_url' => ROOT_URL,
+            'base_url' => $baseUrl,
+            'taobase_www' => $taoBaseWww,
+            'base_www' => $baseWww,
+            'base_lang' => $lang,
+            'locale' => $langCode,
+            'base_authoring_lang' => $this->getUserLanguageService()->getAuthoringLanguage(),
+            'timeout' => $this->getClientTimeout(),
+            'extension' => $resolver->getExtensionId(),
+            'module' => $resolver->getControllerShortName(),
+            'action' => $resolver->getMethodName(),
             'shownExtension' => $this->getShownExtension(),
             'shownStructure' => $this->getShownStructure(),
-            'bundle'         => tao_helpers_Mode::is(tao_helpers_Mode::PRODUCTION)
+            'bundle' => tao_helpers_Mode::is(tao_helpers_Mode::PRODUCTION),
+            'featureFlags' => $this->getFeatureFlagListService()->list(),
         ]));
 
         $this->setView('client_config.tpl');
     }
 
-    /**
-     * Get an extension by it's id
-     * @param string $extensionId the extension name/id
-     * @return common_ext_Extension the extension
-     * @throws Exception if the parameter contains an unknown extension
-     */
-    private function getExtension($extensionId)
-    {
-        try {
-            return $this->getServiceLocator()->get(common_ext_ExtensionsManager::SERVICE_ID)->getExtensionById($extensionId);
-        } catch (common_ext_ExtensionException $cee) {
-            throw new Exception(__('Wrong parameter shownExtension'), $cee);
-        }
-    }
-
     /**
      * @return bool
+     *
      * @throws common_ext_ExtensionException
      */
     protected function isCrossorigin()
     {
-        $ext = $this->getServiceLocator()->get(common_ext_ExtensionsManager::SERVICE_ID)->getExtensionById('tao');
-        $config = $ext->getConfig('js');
-        if ($config != null && isset($config['crossorigin'])) {
+        $config = $this->getExtensionManager()->getExtensionById('tao')->getConfig('js');
+
+        if ($config !== null && isset($config['crossorigin'])) {
             return $config['crossorigin'];
```

Comment:
You can also skip comparing with null since `isset()` will take care of that before trying to access the array index.

That is, `<?php $foo = null; echo isset($foo['bar']) ? 'y' : 'n';` won't result in a warning; but the warning is also not triggered even if `$foo` is not defined, as when doing `<?php echo isset($foo['bar']) ? 'y' : 'n';` directly; I've verified locally just in case :sweat_smile: 

```suggestion
        if (isset($config['crossorigin'])) {
            return $config['crossorigin'];
```

Reply:
I didn't spend a lot of time to refactor old code, but your suggestion is very good, thanks a lot!
Will do this soon 🙂

Reply:
I did in a bit different way 🙂 
Thanks for your comment, fixed!